processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -275,13 +275,10 @@ public DateTime getMaxIngestedEventTime()
       if (preFilters.size() == 0) {
         offset = new NoFilterOffset(0, index.getNumRows(), descending);
       } else {
-        List<ImmutableBitmap> bitmaps = Lists.newArrayList();
-        for (Filter prefilter : preFilters) {
-          bitmaps.add(prefilter.getBitmapIndex(selector));
-        }
+        // Use AndFilter to intersect the preFilters to get its short-circuiting behavior.
         offset = new BitmapOffset(
             selector.getBitmapFactory(),
-            selector.getBitmapFactory().intersection(bitmaps),
+            new AndFilter(preFilters).getBitmapIndex(selector),
```

Comment:
Maybe move the filter intersection logic into a static method (belonging to `AndFilter`, for example, and call it from here and from inside `AndFilter. getBitmapIndex()`? Because creating a dandling instance looks confusing (somebody forgot to create a variable?) to people and to static analysis tools.